util/ValueSetHelper.ts

```diff
@@ -0,0 +1,179 @@
+import { TestCaseInfo } from '../state/atoms/patientTestCase';
+import fhirpath from 'fhirpath';
+import { parsedCodePaths } from '../util/codePaths';
+import { DataRequirementsLookupByTypeProps } from '../state/selectors/dataRequirementsLookupByType';
+import { CodeableConcept } from 'fhir/r2';
+
+export interface GetValueSetCodesProps {
+  code?: string;
+  system?: string;
+}
+
+/**
+ * Helper function that takes a ValueSet url and a measure bundle and returns an
+ * array of objects that contain each code and system in the ValueSet
+ */
+function getValueSetCodes(valueSetUrl: string[], mb: fhir4.Bundle | null): GetValueSetCodesProps[] {
+  const codesAndSystems: GetValueSetCodesProps[] = [];
+  valueSetUrl.forEach(vs => {
+    const vsResource = mb?.entry?.filter(r => r.resource?.resourceType === 'ValueSet' && r.resource?.url === vs)[0]
+      .resource as fhir4.ValueSet;
+    vsResource.expansion?.contains?.forEach(c => {
+      codesAndSystems.push({ code: c.code, system: c.system });
+    });
+  });
+  return codesAndSystems;
+}
+
+/**
+ * Helper function for repeated logic in minimizeTestCaseResources that takes a primaryCodeValue,
+ * a resource, a matching DataRequirement of the same resourceType, a measureBundle, and a newResources
+ * array, checks that the primaryCodeValue exists and if the primaryCodeValue matches any of the
+ * directCodes or ValueSet codes of a DataRequirement of the same resourceType, the resource is added
+ * to the newResources array
+ */
+function checkCodesAndValueSets(
```

Comment:
This function could be simplified to not deal with the `resource` and `newResources` at all. All this needs to do the code comparison and return a boolean if it matches or not. This would retain separation of concerns by only having the `newResources` be managed by the `minimizeTestCaseResources` function.